packages/next/src/build/swc/index.ts

```diff
@@ -861,8 +831,24 @@ function bindingToApi(
                 /* reactCompilerExclude */ undefined
               ),
             ],
-          }
-        }
+            condition: {
+              all: [
+                'browser',
+                { not: 'foreign' },
+                {
+                  content:
+                    options.compilationMode === 'annotation'
+                      ? /['"]use memo['"]/
+                      : !options.compilationMode ||
+                          options.compilationMode === 'infer'
+                        ? // Matches declaration or useXXX or </ (closing jsx) or /> (self closing jsx)
+                          /['"]use memo['"]|\Wuse[A-Z]|<\/|\/>/
+                        : undefined,
+                },
+              ],
+            },
+          },
```

Comment:
should this be structured as looking for `use memo` or matching on file extension .tsx,.jsx?